airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/api.py

```diff
@@ -93,6 +93,9 @@ def handle_call_rate_limit(self, response, params):
             max_pause_interval = self.pause_interval_minimum
 
             for record in response.json():
+                # in case it is failed inner request the headers might not be present
```

Comment:
```suggestion
                # there are two types of failures: 
                # 1. no response (we execute batch until all inner requests has response)
                # 2. response with error (we crash loudly)
                # in case it is failed inner request the headers might not be present
```